control/config.go

```diff
@@ -55,6 +55,7 @@ type config struct {
 	Plugins *pluginConfig `json:"plugins"`
 }
 
+// NewConfig returns a reference to a global config type for the snap daemon
```

Comment:
Would want other people to comment, but maybe something about how it initializes plugin config. It does a pass a reference, but mores specifically it's creating a config struct for someone to add config data to. 
@danielscottt @geauxvirtual